internal/fs/fs_test.go

```diff
@@ -773,33 +768,47 @@ func TestIsEmpty(t *testing.T) {
 	defer h.Cleanup()
 
 	h.TempDir("empty")
-	tests := map[string]string{
-		wd:                                                  "true",
-		"testdata":                                          "true",
-		filepath.Join(wd, "fs.go"):                          "err",
-		filepath.Join(wd, "this_file_does_not_exist.thing"): "false",
-		h.Path("empty"):                                     "false",
-	}
 
-	for f, want := range tests {
-		empty, err := IsNonEmptyDir(f)
-		if want == "err" {
-			if err == nil {
-				t.Fatalf("Wanted an error for %v, but it was nil", f)
-			}
-			if empty {
-				t.Fatalf("Wanted false with error for %v, but got true", f)
+	testCases := []struct {
+		path  string
+		empty bool
+		err   bool
+	}{
+		{wd, true, false},
+		{"testdata", true, false},
+		{filepath.Join(wd, "fs.go"), false, true},
+		{filepath.Join(wd, "this_file_does_not_exist.thing"), false, false},
+		{h.Path("empty"), false, false},
+	}
+
+	// This test case doesn't work on Microsoft Windows because of the
+	// differences in how file permissions are implemented.
+	if runtime.GOOS != "windows" {
```

Comment:
@sdboyer We would like your input on this test case. @ibrasho mentioned that usually an entire test case is skipped for an OS if it is not compatible. This particular test has cases that are all compatible for all operating systems except for the `inaccessibleDir` because of windows file permissions. Your thoughts?